### PR TITLE
BEI-17: Get user profile

### PR DESCRIPTION
**Pull request by hizikex**
#### WHY THIS PR?
There is need for users to see other users' profiles.

#### WHAT DOES THIS PR DO?
It allows users to see other profiles, when they are logged in.

#### HOW CAN THIS PR BE TESTED?
Error will arise when requests is made by without token.

THINGS YOU MAY WANT THE REVIEWER TO NOTE.
N/A

### CODE REVIEW

File: src/api/Route/user.ts [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
 import {authMiddleware}  from '../middleware/authenticate';
 import * as RequestHandler from '../requestHandlers/user';
 
 const router = Router();
 
 router.post('/', RequestHandler.register);
 router.post('/login', RequestHandler.login);
-router.get('/',authMiddleware, getCurrentUser)
+router.get('/',authMiddleware, getCurrentUser);
+router.get('/profile/:username', authMiddleware, getUserProfile);
```
Comment (RICHYDOS) [inline src/api/Route/user.ts:11]: This should be in its own  profile.ts file

**Review by RICHYDOS — CHANGES_REQUESTED** [review]

File: src/core/controllers/profile.ts [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import { ProfileResponse } from '../interfaces/profile';
 import UserRepository from '../repositories/User';
 import FollowerRepository from '../repositories/Follower';
+import {Request, Response, NextFunction } from 'express';
```
Comment (RICHYDOS) [inline src/core/controllers/profile.ts:4]: Remove this line since it's not being used

File: src/core/controllers/user.ts [hunk 3]
```diff
@@ -77,3 +77,23 @@ export const getCurrentUser = async (req: Request, res: Response, next: NextFunc
...
+      const error = new Error('notfound')
+      const resourceNotFoundError = new ResourceNotFoundError("User not found", error, { user })
+      throw resourceNotFoundError
+    }
+    return res.json(user);
+  } catch (err) {
+    next()
+    return
+  }
+};
```
Comment (RICHYDOS) [inline src/core/controllers/user.ts:99]: Remove this method since you moved it to the profile.ts file

File: src/core/interfaces/profile.ts [hunk 4]
```diff
@@ -4,4 +4,3 @@ export interface ProfileResponse {
   image?: string;
   following: boolean;
 }
-
```
Comment (RICHYDOS) [inline src/core/interfaces/profile.ts:7]: is that an underscore?

**Review by RICHYDOS — CHANGES_REQUESTED** [review]

**Comment by LucianaDera** [comment]
Please attend to the comments

File: src/core/interfaces/profile.ts (hunk 4 above)
Comment (hizikex) [reply src/core/interfaces/profile.ts:7]: No

File: src/core/controllers/user.ts (hunk 3 above)
Comment (RICHYDOS) [reply src/core/controllers/user.ts:99]: You did not remove the getProfile controller from the file

**Review by RICHYDOS — CHANGES_REQUESTED** [review]

**Review by RICHYDOS — APPROVED** [review]

**Pull request merged**